homeassistant/components/light/mqtt.py

```diff
@@ -275,10 +320,41 @@ def rgb_received(topic, payload, qos):
         if self._topic[CONF_RGB_COMMAND_TOPIC] is not None:
             self._hs = (0, 0)
 
+        @callback
+        def hue_received(topic, payload, qos):
+            """Handle new MQTT messages for hue."""
+            hue_value = float(templates[CONF_HUE](payload))
+            percent_hue = hue_value / self._hue_scale
+            self._hs = (int(percent_hue * 360), self._hs[1])
+            self.async_schedule_update_ha_state()
+
+        if self._topic[CONF_HUE_STATE_TOPIC] is not None:
+            yield from mqtt.async_subscribe(
+                self.hass, self._topic[CONF_HUE_STATE_TOPIC],
+                hue_received, self._qos)
+            self._hs = (0, 0)
+        if self._topic[CONF_HUE_COMMAND_TOPIC] is not None:
+            self._hs = (0, 0)
+
+        @callback
+        def saturation_received(topic, payload, qos):
+            """Handle new MQTT messages for saturation."""
+            saturation_value = float(templates[CONF_SATURATION](payload))
+            self._hs = (self._hs[0], int(saturation_value))
+            self.async_schedule_update_ha_state()
+
+        if self._topic[CONF_SATURATION_STATE_TOPIC] is not None:
+            yield from mqtt.async_subscribe(
+                self.hass, self._topic[CONF_SATURATION_STATE_TOPIC],
+                saturation_received, self._qos)
+            self._hs = (0, 0)
+        if self._topic[CONF_SATURATION_COMMAND_TOPIC] is not None:
+            self._hs = (0, 0)
+
         @callback
         def color_temp_received(topic, payload, qos):
             """Handle new MQTT messages for color temperature."""
-            self._color_temp = int(templates[CONF_COLOR_TEMP](payload))
+            self._color_temp = int(float(templates[CONF_COLOR_TEMP](payload)))
```

Comment:
? If we don't use floating point numbers internally, why accept floating point numbers? It would IMO just cause confusion because some users would expect us to handle color temp output in floating point to numbers too.

Reply:
This can be resolved by template, so I will delete it.

Thank you for all your review